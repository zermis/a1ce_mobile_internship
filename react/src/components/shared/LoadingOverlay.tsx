import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

import Colors from '../../constants/Colors';

const LoadingOverlay = (): JSX.Element => {
    return (
        <View style={styles.container}>
            <ActivityIndicator size="large" color={Colors.loadingIcon} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        padding: 24
    }
});

export default LoadingOverlay;
