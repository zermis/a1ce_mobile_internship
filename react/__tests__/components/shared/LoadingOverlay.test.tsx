import renderer from 'react-test-renderer';
import LoadingOverlay from '../../../src/components/shared/LoadingOverlay';

it('render loading overlay', () => {
    const tree = renderer.create(<LoadingOverlay />).toJSON();
    expect(tree).toMatchSnapshot();
});
