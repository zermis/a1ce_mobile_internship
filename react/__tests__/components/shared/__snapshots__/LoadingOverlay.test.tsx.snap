// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render loading overlay 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#A53638"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
